server/server.go

```diff
@@ -0,0 +1,77 @@
+package server
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"net"
+	"net/http"
+)
+
+type app struct {
+	server   *http.Server
+	listener *TrackingListener
+}
+
+func newApp(server *http.Server) *app {
+	return &app{
+		server: server,
+	}
+}
+func (a *app) Listener() net.Listener {
+	return a.listener
+}
+
+func (a *app) wait() {
+	a.listener.WaitForChildren()
+}
+
+func (a *app) serve() {
+	a.server.Serve(a.listener)
+}
+
+func (a *app) listen() error {
+	l, err := Listen(a.server.Addr)
+	if err != nil {
+		return err
+	}
+
+	a.listener = NewTrackingListener(l)
+	return nil
+}
+
+func (a *app) signalHandler() {
+
+	c := make(chan os.Signal)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGHUP)
+	go func(c chan os.Signal, listener net.Listener) {
+		for {
+			<-c // os.Signal
+			log.Println("Closing listener")
```

Comment:
Maybe let the user of the library pass a logger so he can hide those messages if he wants.